File: LivrusVirtus/wxWindows/wxRuby/ConAssuntos.rb
require 'wxruby'
require 'Consultas'
require 'resource'
require 'RotinasGlobais'
require 'ConsultasSQL'

class FrmConAssuntos < FrmConsultas
  def initialize(parent)
    super(parent) 
    set_title(Resource::STR_CONASSUNTO)
    @labDescricao.set_label('Assunto:')
    @gridConsulta.set_col_label_value(1, "Assunto")

    evt_button(ID_FRMCONSULTAS_BTNPESQUISAR) {|event| OnBtnPesquisarClick(event)}
    evt_button(ID_FRMCONSULTAS_BTNLIMPAR) {|event| OnBtnLimparClick(event)}
    evt_button(ID_FRMCONSULTAS_BTNFECHAR) {|event| OnBtnFecharClick(event)}
    evt_close {|event| on_close(event)}
    evt_grid_select_cell {|event| OnGridSelectCell(event)}
    evt_grid_label_left_click {|event| OnGridLabelLeftClick(event)}
  end

  def on_close(event)
    self.Fecha(Resource::MNU_CON_ASSUNTOS)
    event.skip()
  end

  def OnBtnPesquisarClick(event)
    self.PesquisarDados(
      $ConsSQL.Assunto('S', "", @edtDescricao.get_value()),
      Resource::STR_ASSNENC, 2)
    event.skip()
  end

  def OnBtnLimparClick(event)
    self.LimparDados()
    event.skip()
  end

  def OnBtnFecharClick(event)
    self.Fecha(0)
    event.skip()
  end

  def OnGridSelectCell(event)
    self.setLabRegistros(event.get_row())
    event.skip()
  end
  
  def OnGridLabelLeftClick(event)
    self.setLabRegistros(event.get_row())
    event.skip()
  end

end

File: LivrusVirtus/VCPP6/wxWin/LibFireBird/ConAssuntos.cpp
#include "resource.h"
#include "Consultas.h"
#include "Principal.h"
#include "ConsultasSQL.h"
#include "RotinasGlobais.h"
#include "ConAssuntos.h"

BEGIN_EVENT_TABLE(CfrmConAssuntos, CfrmConsultas)
  EVT_BUTTON(ID_FRMCONSULTAS_BTNPESQUISAR, CfrmConAssuntos::OnBtnPesquisar)
  EVT_BUTTON(ID_FRMCONSULTAS_BTNLIMPAR, CfrmConAssuntos::OnBtnLimpar)
  EVT_BUTTON(ID_FRMCONSULTAS_BTNFECHAR, CfrmConAssuntos::OnBtnFechar)
  EVT_CLOSE(CfrmConAssuntos::OnClose)
  EVT_GRID_SELECT_CELL(CfrmConAssuntos::OnGridSelectCell)
  EVT_GRID_LABEL_LEFT_CLICK(CfrmConAssuntos::OnGridLabelLeftClick)
END_EVENT_TABLE()

CfrmConAssuntos::CfrmConAssuntos(wxWindow *parent)
       : CfrmConsultas(parent) {
	IniciaControles();
}

CfrmConAssuntos::~CfrmConAssuntos() { }

void CfrmConAssuntos::IniciaControles() {
 this->SetTitle(STR_CONASSUNTO);
 this->labDescricao->SetLabel(wxT("Assunto:  "));
 this->gridConsulta->SetColLabelValue(1, wxT("Assunto"));
}

void CfrmConAssuntos::OnClose(wxCloseEvent& event) {
  Fecha(MNU_CON_ASSUNTOS);
  event.Skip();
}

void CfrmConAssuntos::OnBtnFechar(wxCommandEvent& event) {  
  Fecha(0);
  event.Skip();
}

void CfrmConAssuntos::OnBtnLimpar(wxCommandEvent& event) {
  LimparDados();
  event.Skip();
}

void CfrmConAssuntos::OnBtnPesquisar(wxCommandEvent& event) {
  PesquisarDados(
    ConsSQL->Assunto('S',wxT(""), edtDescricao->GetValue()), STR_ASSNENC, 2);
  event.Skip();
}

void CfrmConAssuntos::OnGridSelectCell(wxGridEvent& event) {
  setLabRegistros(event.GetRow());
  event.Skip();
}

void CfrmConAssuntos::OnGridLabelLeftClick(wxGridEvent& event){
  setLabRegistros(event.GetRow());
  event.Skip();
}

